### DISCUSSION TITLE
Add width to polygon

### DISCUSSION THREAD

**Discussion by tengerye** [python-pillow/Pillow#5687] [Ideas]
May I ask why there is no `width` parameter for the `polygon()`?

**Comment by radarhere** [comment]
No reason that I'm aware of. It simply hasn't been implemented or requested.

**Reply by tengerye** [reply]
@radarhere Thank you for the kind reply. Do you know the proper way to raise such request please? I could only find reporting bugs or something in the `issues`.

**Reply by radarhere** [reply]
Creating an issue is the right way to submit a request.

**Reply by radarhere** [reply]
This became #5689

**Reply by radarhere** [reply]
PR #5694 has been created to resolve this.

**Reply by radarhere** [reply]
This functionality will be part of Pillow 9.0, due out on January 2

**Reply by zhengyuan-public** [reply]
Hi, I'm running Pillow 9.2.0, but it seems the width parameter is still not working. 

`TypeError: polygon() got an unexpected keyword argument 'width'`

**Reply by radarhere** [reply]
That should not be the case. Could you run the following code?
```python
from PIL import Image, ImageDraw
print("Pillow version:", Image.__version__)
im = Image.new("RGB", (1, 1))
draw = ImageDraw.Draw(im)
draw.polygon([(0, 0), (1, 1)], width=5)
```
If it doesn't print a Pillow version of 9.0 or higher, then that is your problem.
If it does print a Pillow version of 9.0 or higher, and my code works, but your code doesn't, then the problem is in your code.

**Comment by RoshanRane** [comment]
I am using Pillow version 9.4.0 and the `width` feature isn't supported for `regular_polygon()`

> TypeError: regular_polygon() got an unexpected keyword argument 'width'

**Reply by radarhere** [reply]
Hi. No, that functionality isn't part of `regular_polygon()`.

For an immediate solution, you should be able to workaround this with

```python
from PIL import Image, ImageDraw
im = Image.new("RGBA", (100, 100))
draw = ImageDraw.Draw(im)

bounding_circle = (50, 50, 50)
n_sides = 5
rotation = 0
fill = "#f00"
outline = "#000"
width = 10
xy = ImageDraw._compute_regular_polygon_vertices(bounding_circle, n_sides, rotation)
draw.polygon(xy, fill, outline, width)
im.show()
```
![tmpvk1g6941](https://user-images.githubusercontent.com/3112309/235915271-6d62878f-ecc7-424b-a458-84b40d9e1eda.PNG)

Are you requesting that `width` be an argument in `regular_polygon()` in Pillow moving forward?

**Reply by RoshanRane** [reply]
Thank you, your workaround works well for my case 😄
Yes, it would be nice to have the same arguments / features in `regular_polygon()` and `polygon()` since the former is just a derivative of the latter.

**Reply by radarhere** [reply]
Ok, I've created PR #7132

**Reply by radarhere** [reply]
#7132 was merged, so it should be part of Pillow 10, due out on July 1.
